### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by marcelveldt**
fix for removing a node (dev_id not constructed the same)
fix for values already cleaned up while hass trying to get unique_id in removal.
Unfortunately still doesn't fix the unique_id issue when the integration gets reloaded. entities stay alive somewhere/somehow.

### CODE REVIEW

File: custom_components/zwave_mqtt/__init__.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def async_node_removed(node):
         # entities itself are removed by the values logic
         if node.id in removed_nodes:
             hass.async_create_task(handle_remove_node(hass, node.id))
-            removed_nodes.remove(node.id)
+            removed_nodes.remove(node)
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/__init__.py]: This doesn't look correct.

File: custom_components/zwave_mqtt/__init__.py [hunk 2]
```diff
@@ -236,11 +237,12 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
...
 
 
-async def handle_remove_node(hass: HomeAssistant, node_id: int):
+async def handle_remove_node(hass: HomeAssistant, node: OZWNode):
     """Handle the removal of a Z-Wave node, removing all traces in device/entity registry."""
     dev_registry = await get_dev_reg(hass)
     # grab device in device registry attached to this node
-    device = dev_registry.async_get_device([(DOMAIN, node_id)], [])
+    dev_id = create_device_id(node)
+    device = dev_registry.async_get_device([(DOMAIN, dev_id)], [])
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/__init__.py]:
```suggestion
    device = dev_registry.async_get_device({(DOMAIN, dev_id)}, set())
```

File: custom_components/zwave_mqtt/entity.py [hunk 3]
```diff
@@ -189,7 +189,7 @@ def device_info(self):
         }
         # device with multiple instances is split up into virtual devices for each instance
         if node_instance > 1:
-            parent_dev_id = f"{ozw_instance}.{node.node_id}.1"
+            parent_dev_id = create_device_id(node, 1)
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/entity.py]: 1 is the default for node_instance.
```suggestion
            parent_dev_id = create_device_id(node)
```

File: custom_components/zwave_mqtt/__init__.py (hunk 1 above)
Comment (marcelveldt) [reply custom_components/zwave_mqtt/__init__.py]: That's what you get when you are too quick and that's why we have reviews ;-)
Thanks

**Review by MartinHjelmare — APPROVED** [review]
Great!

**Pull request merged**